### ISSUE TITLE
std::unordered_multimap key type is string,no match for ‘operator[]’

### ISSUE THREAD

**Issue by wpzybyq** [bytedeco/javacpp-presets#915]
my java configuration file has code as below:
infoMap.put(new Info("std::unordered_multimap<std::string,modsecurity::VariableValue*,modsecurity::MyHash,modsecurity::MyEqual>").pointerTypes("MyMap").define());

generated java file as below:
 @Index public native VariableValue get(@StdString BytePointer i);
    public native MyMap put(@StdString BytePointer i, VariableValue value);

but cpp file compile error:
Java_org_bytedeco_modsecurity_MyMap_get->no match for ‘operator[]’ (operand types are ‘std::unordered_multimap<std::__cxx11::basic_string<char>, modsecurity::VariableValue*, modsecurity::MyHash, modsecurity::MyEqual>’ and ‘std::__cxx11::basic_string<char>’)

Java_org_bytedeco_modsecurity_MyMap_put__Lorg_bytedeco_javacpp_BytePointer_2Lorg_bytedeco_modsecurity_VariableValue_2->no match for ‘operator[]’ (operand types are ‘std::unordered_multimap<std::__cxx11::basic_string<char>, modsecurity::VariableValue*, modsecurity::MyHash, modsecurity::MyEqual>’ and ‘std::__cxx11::basic_string<char>’)

**Comment by saudet** [comment]
That container doesn't look similar to others like std::map. We'll probably need to map it manually like this: https://github.com/bytedeco/javacpp/wiki/Interface-Thrust-and-CUDA

**Comment by wpzybyq** [comment]
> That container doesn't look similar to others like std::map. We'll probably need to map it manually like this: https://github.com/bytedeco/javacpp/wiki/Interface-Thrust-and-CUDA

oh,is there any other examples?just use Thrust-and-CUDA to manually map the code?

**Comment by saudet** [comment]
That's an example for Thrust, when not using the parser, and we can do the same here. We can also try to use the parser on the header files for std::unordered_multimap, but it's not "basic/containers".

**Comment by wpzybyq** [comment]
> That's an example for Thrust, when not using the parser, and we can do the same here. We can also try to use the parser on the header files for std::unordered_multimap, but it's not "basic/containers".

how to use the parser on the header files for std::unordered_multimap?

**Comment by saudet** [comment]
Find a simple implementation of that template, and try to run the parser on it, like this one here:
https://en.cppreference.com/w/cpp/header/unordered_map

**Comment by wpzybyq** [comment]
> Find a simple implementation of that template, and try to run the parser on it, like this one here:
> https://en.cppreference.com/w/cpp/header/unordered_map

@saudet sorry,i haven't understood what you saied."try to run the parser on it",how to run the parser?and who is the parser,javacpp or g++ compiler?

**Comment by saudet** [comment]
By executing the build on a class like this: https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#introduction

**Comment by wpzybyq** [comment]
> By executing the build on a class like this: https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#introduction

@saudet 

when i write my configuration class 'UnSortedMapConfig.java' as below
`import org.bytedeco.javacpp.*;
import org.bytedeco.javacpp.annotation.*;
import org.bytedeco.javacpp.tools.*;

@Properties(
        value = @Platform(
                includepath = {"/opt/gcc-5.2.0/libstdc++-v3/include/"},
                include = {"/opt/gcc-5.2.0/libstdc++-v3/include/bits/unordered_map.h"},
                link = {"UnSortedMap"}
        ),
        target = "UnSortedMap"
)
public class UnSortedMapConfig implements InfoMapper {
        public void map(InfoMap infoMap) {
        }
}
`

and executing the build as below 
`java -jar javacpp.jar UnSortedMapConfig.java `

get error:
`Exception in thread "main" org.bytedeco.javacpp.tools.ParserException: /opt/gcc-5.2.0/libstdc++-v3/include/bits/unordered_map.h:33:Unexpected token '_GLIBCXX_VISIBILITY'`

the unordered_map.h header file has comment as below:
`@file bits/unordered_map.h
 *  This is an internal header file, included by other library headers.
 *  Do not attempt to use it directly. @headername{unordered_map}`

does this header file not allow to be build seperately?

**Comment by saudet** [comment]
Yes, like that, but the GCC implementation is pretty complicated.
Like I said, it would be easier with a simpler implementation like the one here:
https://en.cppreference.com/w/cpp/header/unordered_map

**Comment by wpzybyq** [comment]
> Yes, like that, but the GCC implementation is pretty complicated.
> Like I said, it would be easier with a simpler implementation like the one here:
> https://en.cppreference.com/w/cpp/header/unordered_map

@saudet sorry,i don't know how to start a simpler implementation,can you show me some examples?

**Comment by saudet** [comment]
I gave you the link twice! What is it that you do not understand?

**Comment by wpzybyq** [comment]
this link shows 

> I gave you the link twice! What is it that you do not understand?

@saudet i'm so sorry.as i know,users give java configuration file,and then javacpp auto gennerate cpp file and build jni file,so i haven't understood how to use this link combine with javacpp.

**Comment by wpzybyq** [comment]
> this link shows
> 
> > I gave you the link twice! What is it that you do not understand?
> 
> @saudet i'm so sorry.as i know,users give java configuration file,and then javacpp auto gennerate cpp file and build jni file,so i haven't understood how to use this link combine with javacpp.

at first,i think using the command 'java -jar javacpp.jar xxConfig.java' to generate the special class(for example:unordered_multimap),but get the compile error.so i cufused that this link tells me how to solve this compile error or another method combine with javacpp?

**Comment by saudet** [comment]
If you'd like to continue and try to use the one from GCC, we can fix that error this way:
https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#redefining-the-code-of-a-macro

**Comment by wpzybyq** [comment]
> If you'd like to continue and try to use the one from GCC, we can fix that error this way:
> https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes#redefining-the-code-of-a-macro

thank you for your reply. i will hava a try on this.
